core/src/main/java/io/questdb/std/str/Utf8s.java

```diff
@@ -914,21 +977,33 @@ public static void trim(TrimType type, Utf8Sequence source, Utf8Sink sink) {
      * @return an integer-encoded tuple (decoded number of bytes, character in UTF-16 encoding, stored as short type)
      */
     public static int utf8CharDecode(Utf8Sequence seq) {
-        int size = seq.size();
+        return utf8CharDecode(seq, 0);
+    }
+
+    /**
+     * A specialised function to decode a single UTF-8 character.
+     * Used when it doesn't make sense to allocate a temporary sink.
+     *
+     * @param seq    input sequence
+     * @param offset offset into the sequence
+     * @return an integer-encoded tuple (decoded number of bytes, character in UTF-16 encoding, stored as short type)
+     */
+    public static int utf8CharDecode(Utf8Sequence seq, int offset) {
```

Comment:
Some UTF-8 codes correspond to a surrogate pair of UTF-16 codes. Since there's nowhere to store a 32-bit surrogate pair in this packed `int`, it seems this method will break on a surrogate pair.